### PR TITLE
Add firebaseExecutors extension methods for Tests

### PR DESCRIPTION
**Pull request by daymxn**
Per [b/279937403](https://b.corp.google.com/issues/279937403),

This provides extension methods to the `TestScope` to better facilitate `CoroutineContext` usage in tests; allowing product teams to write tests with Firebase's `TestOnlyExecutors` that support `TestScope` features such as fast-forwarding.

Additionally, this does + fixes the following:

- [b/279949209](https://b.corp.google.com/issues/279949209) -> Migrate `integ-testings` to KTS + VC
- [b/279935610](https://b.corp.google.com/issues/279935610) -> Update Executor documentation to include edge case scenarios, and better articulate usage not only across tests- but general Kotlin usage as well

### CODE REVIEW

File: contributor-docs/components/executors.md [hunk 1]
```diff
@@ -177,20 +177,50 @@ Qualified<Executor> bgExecutor = qualified(Background.class, Executor.class);
...
 
-To do that, you should prefer relying on Components to inject the right executor even in tests. This will ensure
-your tests are always using the executor that is actually used in your SDK build.
-If your SDK uses Dagger, see [Dependency Injection]({{ site.baseurl }}{% link best_practices/dependency_injection.md %})
+To do that, you should prefer relying on Components to inject the right executor
+even in tests. This will ensure
+your tests are always using the executor that is actually used in your SDK
+build.
+If your SDK uses Dagger, see [Dependency Injection]({{ site.baseurl }}{% link
+best_practices/dependency_injection.md %})
```
[the remark this answers is not in the document]
Comment (rlazo) [reply contributor-docs/components/executors.md:219]: friendly ping

File: integ-testing/integ-testing.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+plugins {
+  id("com.android.library")
+  kotlin("android")
+}
+
+android {
+  val targetSdkVersion: Int by rootProject
+  val minSdkVersion: Int by rootProject
+
+  kotlinOptions.jvmTarget = "1.8"
```
[the remark this answers is not in the document]
Comment (daymxn) [reply integ-testing/integ-testing.gradle.kts]: I don't believe so... lets find out ;)

File: contributor-docs/components/executors.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (daymxn) [reply contributor-docs/components/executors.md]: ktfmt hard at work 👍

File: contributor-docs/components/executors.md [hunk 3]
```diff
@@ -177,20 +177,50 @@ Qualified<Executor> bgExecutor = qualified(Background.class, Executor.class);
...
 and [Dagger's testing guide](https://dagger.dev/dev-guide/testing).
 
-When the above is not an option, you can use `TestOnlyExecutors`, but make sure you're testing your code with
+When the above is not an option, you can use `TestOnlyExecutors`, but make sure
+you're testing your code with
 the same executor that is used in production code:
```
[the remark this answers is not in the document]
Comment (daymxn) [reply contributor-docs/components/executors.md]: Fixed
